.github/workflows/release.yml

```diff
@@ -14,6 +14,10 @@ on:
 
 jobs:
   docker:
+    # Need's repo owner login creds
+    if: |
+      (github.event_name == 'pull_request' && github.event.pull_request.user.login == github.repository_owner) ||
+      github.actor == github.repository_owner
```

Comment:
rather than conditionally running the job, you can conditionally run the login action and dynamically set the push field

Reply:
The docker job does nothing that the `make test` does not cover. What does building a container and not pushing it achieve?

Reply:
* Regarding the first point, testing that a container builds is not very useful when all it does is copy the static binary over built from what `make test` builds.

https://github.com/adyanth/cloudflare-operator/blob/2aed182f375ed782904e54337ff7650f88595ec1/Dockerfile#L28-L33

* Regarding the build failures, this change would prevent exactly that right? By not running the job. The `test` job would still run.

Open to changing it after seeing how it behaves once it is merged to main.

Reply:
In this instance it's definitely a small difference, but it also means this being built is actually running in CI. I'm DevOps by trade and a firm believer that more checks is more good haha